core/src/main/java/org/apache/iceberg/BaseFilesTable.java

```diff
@@ -46,15 +45,37 @@ abstract class BaseFilesTable extends BaseMetadataTable {
 
   @Override
   public Schema schema() {
-    StructType partitionType = Partitioning.partitionType(table());
+    // avoid returning an empty struct, which is not always supported.
```

Comment:
I am a bit lost with all the schemas here. Let me see if we can simplify this block.